src/Connection.js

```diff
@@ -0,0 +1,52 @@
+
+define(["ObjectsConfig"], function (ObjectsConfig) {
+    "use strict";
+
+    var Factory = function () {
+        this.createConnection = function (url) {
+            var session = new ObjectsConfig.WebSocket(url);
+            var connection = new Connection(session);
+            return connection;
+        }
+    }
+
+    var Connection = function(session) {
+        var _session = session;
+
+        this.getSession = function () {
+            return _session;
+        }
+
+        _session.onopen = function (e) {
+            console.log("connection on " + session.url + " started.");
+        }
+
+        _session.onclose = function (e) {
+            console.log("connection on " + session.url + " closed.");
+        }
+
+        _session.onerror = function (e) {
+            console.error("Error on " + session.url + ".");
+        }
+
+    }
+
```

Comment:
the connection should only have an asynchronous method CreateSession with a callback parameter (error, session)